src/Algebra/Ordered/Structures.agda

```diff
@@ -0,0 +1,308 @@
+  field
+    isPromonoid : IsPromonoid ∙ ε
+    comm        : Commutative ∙
+
+  open IsPromonoid isPromonoid public
+
+  isCommutativeMonoid : IsCommutativeMonoid ∙ ε
+  isCommutativeMonoid = record { isMonoid = isMonoid ; comm = comm }
+
+  open IsCommutativeMonoid isCommutativeMonoid public
+    using (isCommutativeSemigroup)
+
+-- Preordered semirings (prosemirings)
+
+record IsProsemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
+  field
+    +-isCommutativePromonoid : IsCommutativePromonoid + 0#
+    *-cong                   : Congruent₂ *
+    *-mono                   : Monotonic₂ *
+    *-assoc                  : Associative *
+    *-identity               : Identity 1# *
+    distrib                  : * DistributesOver +
+    zero                     : Zero 0# *
+
+  open IsCommutativePromonoid +-isCommutativePromonoid public
+    renaming
+    ( assoc                  to +-assoc
+    ; ∙-cong                 to +-cong
+    ; ∙-congˡ                to +-congˡ
+    ; ∙-congʳ                to +-congʳ
+    ; mono                   to +-mono
+    ; identity               to +-identity
+    ; identityˡ              to +-identityˡ
+    ; identityʳ              to +-identityʳ
+    ; comm                   to +-comm
+    ; isMagma                to +-isMagma
+    ; isSemigroup            to +-isSemigroup
+    ; isMonoid               to +-isMonoid
+    ; isCommutativeSemigroup to +-isCommutativeSemigroup
+    ; isCommutativeMonoid    to +-isCommutativeMonoid
+    )
+
+  *-isPromonoid : IsPromonoid * 1#
+  *-isPromonoid = record
+    { isProsemigroup = record
+      { isPromagma   = record
+        { isPreorder = isPreorder
+        ; ∙-cong     = *-cong
+        ; mono       = *-mono
+        }
+      ; assoc        = *-assoc
+      }
+    ; identity       = *-identity
+    }
+
+  open IsPromonoid *-isPromonoid public
+    using ()
+    renaming
+    ( ∙-congˡ     to *-congˡ
+    ; ∙-congʳ     to *-congʳ
+    ; identityˡ   to *-identityˡ
+    ; identityʳ   to *-identityʳ
+    ; isMagma     to *-isMagma
+    ; isSemigroup to *-isSemigroup
+    ; isMonoid    to *-isMonoid
+    )
+
+  isSemiring : IsSemiring + * 0# 1#
+  isSemiring = record
+    { isSemiringWithoutAnnihilatingZero = record
+      { +-isCommutativeMonoid           = +-isCommutativeMonoid
+      ; *-cong                          = *-cong
+      ; *-assoc                         = *-assoc
+      ; *-identity                      = *-identity
+      ; distrib                         = distrib
+      }
+    ; zero                              = zero
+    }
+
+  open IsSemiring isSemiring public using (distribˡ; distribʳ; zeroˡ; zeroʳ)
+
+------------------------------------------------------------------------
+-- Partially ordered structures
+
+-- Partially ordered magmas (pomagmas)
+
+record IsPomagma (∙ : Op₂ A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
+  field
+    isPartialOrder : IsPartialOrder _≈_ _≤_
+    mono           : Monotonic₂ ∙
```

Comment:
Maybe this should be `∙-mono` to fit with `∙-cong`, but I can see the argument both ways.

Reply:
Yes, I went back and forth on that one as well but settled for the shorter variant as the extra prefix doesn't really add any info when there's just one operation.

Reply:
Yup, let's stick with no prefix. I still hope to convince people to get rid of the `∙-` in `∙-cong` some day.